python/cudf/cudf/tests/test_parquet.py

```diff
@@ -2005,7 +2008,15 @@ def test_parquet_nullable_boolean(tmpdir, engine):
     expected_gdf = cudf.DataFrame({"a": [True, False, None, True, False]})
 
     pdf.to_parquet(pandas_path)
-    actual_gdf = cudf.read_parquet(pandas_path, engine=engine)
+    if engine == "pyarrow":
+        with pytest.warns(
+            UserWarning,
+            match="Using CPU via PyArrow to read Parquet dataset. This option "
+            "is both inefficient and unstable!",
+        ):
+            actual_gdf = cudf.read_parquet(pandas_path, engine=engine)
+    else:
+        actual_gdf = cudf.read_parquet(pandas_path, engine=engine)
```

Comment:
I've been taking a slightly different approach in other PRs addressing warnings. Since this code block is effectively used twice, you can define a context manager at the top of the file and re-use it in both cases. It has the additional benefit of giving the context a very explicit name:

```python
from contextlib import contextmanager

@contextmanager
def _hide_pyarrow_parquet_cpu_warnings(engine):
    if engine == "pyarrow":
        with pytest.warns(
            UserWarning,
            match="Using CPU via PyArrow to read Parquet dataset. This option "
            "is both inefficient and unstable!",
        ):
            yield
    else:
        yield
```

Then:
```suggestion
    with _hide_pyarrow_parquet_cpu_warnings(engine):
        actual_gdf = cudf.read_parquet(pandas_path, engine=engine)
```